### PR TITLE
Allow empty classifier. Fixes #132

### LINKED ISSUE #132: Support stubrunner's dependency with empty classifier
Looks like it isn't possible to have stub defined as `company:service:+::`

### PR DESCRIPTION
**Pull request by dstepanov**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@dstepanov Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-contract&pullRequestId=134)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-contract&pullRequestId=134) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@dstepanov Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-contract&pullRequestId=134)!

File: spring-cloud-contract-stub-runner/src/test/groovy/org/springframework/cloud/contract/stubrunner/StubRunnerExecutorSpec.groovy [hunk 1]
```diff
@@ -95,6 +95,24 @@ class StubRunnerExecutorSpec extends Specification {
 		executor.shutdown()
 	}
 
+	def 'should much stub with empty classifier'() {
```
Comment (marcingrzejszczak) [inline spring-cloud-contract-stub-runner/src/test/groovy/org/springframework/cloud/contract/stubrunner/StubRunnerExecutorSpec.groovy]: I guess you meant `match` :D

File: spring-cloud-contract-stub-runner/src/test/groovy/org/springframework/cloud/contract/stubrunner/StubConfigurationSpec.groovy [hunk 2]
```diff
@@ -17,30 +17,32 @@
...
+			stubConfiguration.artifactId == artifactId
+			stubConfiguration.groupId == groupId
+			stubConfiguration.classifier == classifier
+			stubConfiguration.version == version
+		and:
+			stubConfiguration.toColonSeparatedDependencyNotation() == ivyNotation
+		and:
+			stubConfiguration.matchesIvyNotation(ivy)
+		where:
+			ivy                                 | ivyNotation                         | artifactId | groupId | classifier   | version
```
Comment (marcingrzejszczak) [inline spring-cloud-contract-stub-runner/src/test/groovy/org/springframework/cloud/contract/stubrunner/StubConfigurationSpec.groovy]: can you put a double `||` between `ivy` and `ivyNotation` ?

File: spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/StubRunnerExecutor.java [hunk 3]
```diff
@@ -120,8 +120,8 @@ private boolean classifierMatches(String[] splitString) {
 	}
 
 	private boolean groupIdArtifactVersionMatches(String[] splitString) {
-		return this.stubServer.stubConfiguration.artifactId.equals(splitString[0])
-				&& this.stubServer.stubConfiguration.groupId.equals(splitString[1])
+		return this.stubServer.stubConfiguration.groupId.equals(splitString[0])
```
Comment (marcingrzejszczak) [inline spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/StubRunnerExecutor.java]: Ooops :P

**Comment by dstepanov** [comment]
Typos addressed :)

**Pull request merged**